Improve VDSM readme file

barpavel opened the pull request:
Multiple improvements and fixes to VDSM `README.md` file.

aesteve-rh commented:
This will collide with https://github.com/oVirt/vdsm/pull/147

So we have to take it into account for whoever gets merged last, probably will have conflicts to be solved :)

barpavel commented:
> 

I see that you mostly deleted stuff from that `README.md` and moved them to another place...
In that case maybe it's easier to merge my first and then just delete 2 sections.
You can take or throw away whatever I changed...
Up to you & Nir, you are the VDSM guys :)

aesteve-rh commented:
I'm a VDSM trainee :D, but yes I agree. Also I'll be OoO next Monday, so you'll probably merge first regardless ;)

I'll handle the conflicts, no problem.

nirs commented on README.md:
```diff
@@ -3,10 +3,10 @@
 [![Copr build status](https://copr.fedorainfracloud.org/coprs/ovirt/ovirt-master-snapshot/package/vdsm/status_image/last_build.png)](https://copr.fedorainfracloud.org/coprs/ovirt/ovirt-master-snapshot/package/vdsm/)
 
-Welcome to the Vdsm source repository.
+Welcome to the VDSM source repository.
```
We intentionally do not use VDSM. You can replace "vdsm" with "Vdsm" if needed.

nirs commented on README.md:
```diff
@@ -163,12 +163,12 @@ oVirt's Jenkins CI is the integrated method for testing VDSM patches,
+- Visit https://travis-ci.org, register using your GitHub account, and
   enable builds for your VDSM fork.
-- Push your changes to your github fork to trigger a build.
+- Push your changes to your GitHub fork to trigger a build.
```
The entire section should be removed, since we don't use Travis now, no need to fix it.

nirs commented on README.md:
```diff
@@ -186,7 +186,9 @@ There are two mailing lists for discussions:
-The developers also hang out on IRC at #vdsm hosted on freenode.net
+The developers also hang out on IRC at
+[#vdsm](irc://irc.freenode.org/vdsm) hosted on freenode.net
+([logs](http://ekohl.nl/vdsm)).
```
#vdsm channel is not used for years, and nobody uses freenode now. The link should be replaced by #ovirt on irc.oftc.net.

nirs commented on README.md:
```diff
@@ -144,7 +144,7 @@ steps:
 - Updating releng-tools with the new VDSM version.  See releng-tools
-  repo (`git clone https://gerrit.ovirt.org/releng-tools`) and VDSM
+  repo (`git clone https://github.com/oVirt/releng-tools`) and VDSM
   related patches there for examples.
```
The change is good, but it will be more useful if we link to a previous patch showing showing how the change looks like. Here is how we do this in imageio:
https://github.com/oVirt/ovirt-imageio/blob/master/RELEASE.md#upstream

nirs commented on README.md:
```diff
@@ -154,15 +154,15 @@ Running tests locally is convenient, but before your changes can be
 are not covered currently by Jenkins CI. This can be achieved in a fast
-way manually and independently from gerrit by invoking Travis CI on your
+way manually and independently by invoking Travis CI on your
 GitHub branch:
```
This section should be removed, no need to fix it.

nirs commented on README.md:
```diff
@@ -39,7 +39,7 @@ For CentOS Stream 8 use:
     sudo dnf install -y ovirt-release-master
 
-For more info see
+**Note:** for more info see
```
Why do we need this note?

nirs commented on README.md:
```diff
@@ -198,18 +198,18 @@ The latest upstream code can be obtained from GitHub:
 If you modify this program, you may extend this exception to your
-version, but you are not obligated to do so.  If you do not wish to do
+version, but you are not obligated to do so. If you do not wish to do
 so, delete this exception statement from your version.
```
Removing the 2 spaces at the end is not needed. Quoting names in the entire document looks nice.

nirs commented on README.md:
```diff
@@ -155,7 +155,7 @@ merged, we need to test them on all supported distributions and
 
 When you submit patches to GitHub, oVirt's Jenkins CI will run its tests
-according to configuration in the stdci.yaml file.
+according to configuration in the `stdci.yaml` file.
```
We don't use stdci.yml since we moved to github. This section need to be rewritten instead beautifying stale text.

nirs commented on README.md:
````diff
@@ -35,31 +35,36 @@ date, please check the [Gerrit VDSM project](https://gerrit.ovirt.org/q/project:
+```shell
+$ sudo dnf copr enable -y ovirt/ovirt-master-snapshot centos-stream-8
+$ sudo dnf install -y ovirt-release-master
+```
````
May be this looks better in pycharm but this format is harder to read and edit. Please keep the generic indented block as is.

nirs commented on README.md:
```diff
@@ -15,15 +15,16 @@ log collection.
 
-Please use GitHub pull requests.
+Please use [GitHub pull requests](https://github.com/oVirt/vdsm/pulls).
+See [Development environment setup](#development-environment-setup) for more details.
```
Link is useful, but we want to move the section to a new document, so better wait until this work is done.

nirs commented on README.md:
```diff
@@ -15,15 +15,16 @@ log collection.
-If you find a documentation issue on the oVirt website please navigate
-and click "Report an issue on GitHub" in the page footer.
+If you find a documentation issue on the [oVirt website](https://www.ovirt.org) please navigate
+and click "Report an issue with this page" in the page footer.
```
This text was added by Sandro to many projects - most likely the same change is needed in many other projects.

nirs commented on README.md:
````diff
@@ -88,6 +88,7 @@ $ sudo dnf copr enable -y nsoffer/ovirt-imageio-preview
 Install additional packages for Fedora, CentOS, and RHEL:
 ```shell
+$ cd vdsm
 $ sudo dnf install -y `cat automation/check-patch.packages`
````
I'm not sure the "cd vdsm" part is needed.

The automation directory should be removed and this command should be replaced. I think Albert is working this in another PR.

nirs commented on README.md:
````diff
@@ -120,9 +111,21 @@ $ make
+- When done, you can deactivate the environment:
+  ```shell
+  $ deactivate
+  ```
````
Moving the instructions how to activate and deactivate the virtual environment to the section about running the tests makes sense, but it conflicts with Albert changes. Let's wait until this work is done.

nirs commented:
@barpavel Let's remove all the changes conflicting with #147, and post later if need.

aesteve-rh commented:
@barpavel fyi, https://github.com/oVirt/vdsm/pull/147 has been integrated, so you can already rebase and proceed with this one.

nirs commented:
@barpavel most changes here are already fixed. Please rebase the PR on master.

nirs commented:
@barpavel please close if you don't plan to work on this.

barpavel commented:
I will check later which of the changes are still relevant / worth adding.
